spec/lib/debt_management_center/statement_identifier_service_spec.rb

```diff
@@ -7,6 +7,194 @@
                type: :service do
   describe '#get_mpi_data' do
     Flipper.disable(:va_v3_contact_information_service)
+
+    context 'given edipi statement' do
+      edipi = '492031291'
+      let(:verification) { build(:dslogon_user_verification) }
+      let(:edipi_statement) do
+        {
+          'veteranIdentifier' => edipi,
+          'identifierType' => 'edipi',
+          'facilityNum' => '123',
+          'facilityName' => 'VA Medical Center',
+          'statementDate' => '01/01/2023'
+        }
+      end
+      let(:mpi_profile) { build(:mpi_profile) }
+      let(:profile_response) { create(:find_profile_response, profile: mpi_profile) }
+
+      context 'when MPI gets a GatewayTimeout' do
+        let(:address) { 'person43@example.com' }
+        let(:expected_error) { Common::Exceptions::GatewayTimeout }
+        let(:expected_error_message) { expected_error.new.message }
+
+        before { allow_any_instance_of(Faraday::Connection).to receive(:post).and_raise(Faraday::TimeoutError) }
+
+        it 'recognizes this error is retryable' do
+          service = described_class.new(edipi_statement)
+
+          expect { service.get_mpi_data }
+            .to raise_error(described_class::RetryableError)
+        end
+      end
+
+      context 'when MPI fails to breakers outage' do
+        let(:current_time) { Time.zone.now }
+        let(:expected_error) { Breakers::OutageException }
+        let(:expected_error_message) { "Outage detected on MVI beginning at #{current_time.to_i}" }
+
+        before do
+          Timecop.freeze
+          MPI::Configuration.instance.breakers_service.begin_forced_outage!
+        end
+
+        after { Timecop.return }
+
+        it 'recognizes this error is retryable' do
+          service = described_class.new(edipi_statement)
+          expect { service.get_mpi_data }.to raise_error(described_class::RetryableError)
+        end
+      end
+
+      context 'MPI profile found' do
+        before do
+          mpi_profile.vet360_id = '1'
```

Comment:
You can define this when you do the `mpi_profile` definition:
`let(:mpi_profile) { build(:mpi_profile, vet360_id: 1) }`

Reply:
(it's a little more reliable than mutating the object after it's been created)

Reply:
if this is all gonna change substantially or go away, this comment can probably be ignored. It's more of a 'nitpick' than anything else